### PR TITLE
Partial materials refactor

### PR DESCRIPTION
**Pull request by CheffieGithub**
## About the PR 
I needed material stuff for a prefab, ended up making furniture use the same material set call instead of 5 different implentations across all the things. Now for objects its set for everything on New() on the base object type.

Fixes wooden tables and the like giving you steel sheets probably.

Test merge this since it touches a lot of material stuff, I tested it myself but can't be sure nothing broke.

## Why's this needed? 
Consistency is good. Easier to make new material stuff.

### CODE REVIEW

**Comment by pali6** [comment]
Good PR at a glance. Will review once it passes checks.

File: code/obj/stool.dm [hunk 1]
```diff
@@ -874,6 +876,23 @@
...
+/* -------------------- Material Chairs ------------------ */
+/* ======================================================= */
+
+/obj/stool/chair/material
+	name = "material chair"
+	desc = "A chair made from a material"
+	uses_material_appearance = TRUE
+	comfort_value = 4
+
+	mauxite
```
Comment (ZeWaka) [inline code/obj/stool.dm:889]: Can you absolutely path this?

**Review by ZeWaka — APPROVED** [review]

File: code/obj/stool.dm [hunk 2]
```diff
@@ -874,6 +876,23 @@
...
+
+/obj/stool/chair/material
+	name = "material chair"
+	desc = "A chair made from a material"
+	uses_material_appearance = TRUE
+	mat_changename = TRUE
+
+/obj/stool/chair/material/mauxite
+	name = "chair"
+	icon_state = "chair$$mauxite"
```
Comment (pali6) [inline code/obj/stool.dm:891]: hmm Is this necessary, I reckon the setMaterial code will replace it with this icon state automatically

File: code/obj/table.dm [hunk 3]
```diff
@@ -532,6 +535,22 @@ TYPEINFO_NEW(/obj/table/scrap)
...
 		auto = 1
 
+TYPEINFO(/obj/table/mauxite)
+TYPEINFO_NEW(/obj/table/mauxite)
+	. = ..()
+	smooth_list = typecacheof(/obj/table/mauxite/auto)
+/obj/table/mauxite
+	name = "table"
+	icon = 'icons/obj/furniture/table.dmi'
+	icon_state = "0$$mauxite"
```
Comment (pali6) [inline code/obj/table.dm:545]: ditto

File: code/obj/table.dm [hunk 4]
```diff
@@ -532,6 +535,22 @@ TYPEINFO_NEW(/obj/table/scrap)
...
+	desc = "A table made of dense mauxite"
```
Comment (pali6) [inline code/obj/table.dm:546]: Won't the description already get changed by the mat to add the mauxite bit? (i.e. isn't this kinda redunant and possibly false if the table gets transmuted later?)

**Review by pali6 — APPROVED** [review]

File: code/obj/stool.dm (hunk 2 above)
Comment (CheffieGithub) [reply code/obj/stool.dm:891]: for map editor sprites

**Pull request merged**